Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/DialogHelper.kt

```diff
@@ -1,141 +0,0 @@
-            positiveButtonFunction: () -> Unit? = {},
-            negativeButtonFunction: () -> Unit? = {},
-            cancelFunction: () -> Unit? = {},
-            isDeleteDialog: Boolean = false
-        ) : this(
-            context = context,
-            title = context.resources.getString(title),
-            message = message,
-            positiveButton = context.resources.getString(positiveButton),
-            negativeButton = negativeButton?.let { context.resources.getString(it) },
-            cancelable = cancelable,
-            positiveButtonFunction = positiveButtonFunction,
-            negativeButtonFunction = negativeButtonFunction,
-            cancelFunction = cancelFunction,
-            isDeleteDialog = isDeleteDialog
-        )
-    }
-
-    fun showDialog(
-        dialogInstance: DialogInstance
-    ): AlertDialog {
-        val message = getMessage(
-            dialogInstance.context,
-            dialogInstance.message,
-            dialogInstance.isTextSelectable
-        )
-
-        val alertDialog = dialogInstance.context.let {
-            val builder = MaterialAlertDialogBuilder(it)
-            builder.apply {
-                setTitle(dialogInstance.title)
-                setView(message)
-                setCancelable(dialogInstance.cancelable ?: true)
-                setPositiveButton(
-                    dialogInstance.positiveButton
-                ) { _, _ ->
-                    dialogInstance.positiveButtonFunction()
-                }
-                if (dialogInstance.negativeButton != null) {
-                    setNegativeButton(
-                        dialogInstance.negativeButton
-                    ) { _, _ ->
-                        dialogInstance.negativeButtonFunction()
-                    }
-                }
-                setOnCancelListener { dialogInstance.cancelFunction() }
-            }
-            builder.create()
-        }
-
-        return alertDialog.apply {
-            show()
-            if (dialogInstance.isDeleteDialog) {
-                this.getButton(DialogInterface.BUTTON_POSITIVE)
-                    ?.setTextColor(dialogInstance.context.getColorCompat(R.color.colorTextDeleteButtonDialog))
-            }
-        }
-    }
-
-    private fun getMessage(context: Context, message: String?, isTextSelectable: Boolean): TextView {
-        // create spannable and add links, removed stack trace links into nowhere
-        val spannable = SpannableString(message)
-        val httpPattern: Pattern = Pattern.compile("[a-z]+://[^ \\n]*")
-        Linkify.addLinks(spannable, httpPattern, "")
-        // get padding for all sides
-        val paddingStartEnd = context.resources.getDimension(R.dimen.spacing_normal).toInt()
-        val paddingLeftRight = context.resources.getDimension(R.dimen.spacing_small).toInt()
-        // create a textview with clickable links from the spannable
-        val textView = TextView(context)
```

Comment:
I think some error dialogs have links, will they be still working ?

Reply:
I'll check.